File: Python/consecutive_strings.py
def longest_consec(strarr, k):
    longest = ""
    i = 0
    while i <= len(strarr) - k:
        consec = ""
        for j in range(0, k):
            consec += strarr[i + j]
        if len(consec) >= len(longest):
            longest = consec
        i += 1
    return longest


print(longest_consec(["zone", "abigail", "theta", "form", "libe", "zas"], 2))
# => "abigailtheta"
print(longest_consec(["ejjjjmmtthh", "zxxuueeg", "aanlljrrrxx",
      "dqqqaaabbb", "oocccffuucccjjjkkkjyyyeehh"], 1))
# => "oocccffuucccjjjkkkjyyyeehh"
print(longest_consec([], 3))
# => ""

File: JavaScript/consecutiveStrings.js
function longestConsec(strarr, k) {
  longest = "";
  for (let i = 0; i <= strarr.length - k; i++) {
    group = "";
    for (let j = i; j < i + k; j++) {
      group += strarr[j];
    }
    if (group.length > longest.length) {
      longest = group;
    }
  }
  return longest;
}

console.log(longestConsec(["zone", "abigail", "theta", "form", "libe", "zas"], 2)); // => "abigailtheta"
console.log(longestConsec(["ejjjjmmtthh", "zxxuueeg", "aanlljrrrxx", "dqqqaaabbb", "oocccffuucccjjjkkkjyyyeehh"], 1)); // => "oocccffuucccjjjkkkjyyyeehh"
console.log(longestConsec([], 3)); // > ""
